api/application/client.go

```diff
@@ -232,20 +232,70 @@ func (c *Client) AddUnits(application string, numUnits int, placement []*instanc
 	return results.Units, err
 }
 
-// DestroyUnits decreases the number of units dedicated to an application.
-func (c *Client) DestroyUnits(unitNames ...string) error {
+// DestroyUnitsDeprecated decreases the number of units dedicated to an
+// application.
+//
+// NOTE(axw) this exists only for backwards compatibility, for API facade
+// versions 1-3; clients should prefer its successor, DestroyUnit, below.
+func (c *Client) DestroyUnitsDeprecated(unitNames ...string) error {
 	params := params.DestroyApplicationUnits{unitNames}
 	return c.facade.FacadeCall("DestroyUnits", params, nil)
 }
 
-// Destroy destroys a given application.
-func (c *Client) Destroy(application string) error {
+// DestroyUnit decreases the number of units dedicated to an application.
+func (c *Client) DestroyUnit(unitNames ...string) ([]params.DestroyUnitResult, error) {
+	args := params.Entities{
+		Entities: make([]params.Entity, len(unitNames)),
+	}
+	for i, name := range unitNames {
+		if !names.IsValidUnit(name) {
+			return nil, errors.NotValidf("unit name %q", name)
```

Comment:
IMO it's OK here because the user has specified something invalid, rather than there being a server-side error for one of the parameters. I'll do it anyway, but question the value.